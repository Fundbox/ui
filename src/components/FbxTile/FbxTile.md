# FbxTile
The general fundbox tile.

Used mainly inside grid.

<br>
<br>

## The `Tile` Constructor
This component exposes a `Tile` class which you should use to construct the tiles.

You will need to pass three type of  `<image/>.src` valid strings to this `Tile` constructor. You must supply normal, hover and disabled images. 
```js
// Use the `Tile` constructor to create the data passed to tile component
const tileData = new Tile(tileName, imageNormal, imageHover, imageDisabled, hoverColor, isEnabled, isConnected, metadata);
```

```html
<fbx-tile :tile="tileData"></fbx-tile>
```
