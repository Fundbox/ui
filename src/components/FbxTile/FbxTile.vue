<template>
  <div>Hello from FbxTile!</div>
</template>

<script>
export default {
  name: 'FbxTile'
}
</script>

<style lang="scss" scoped>
</style>
