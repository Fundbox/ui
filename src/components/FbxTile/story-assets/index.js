export { default as img5Normal } from '../../../assets/story-assets/5_colored.svg'
export { default as img5Hover } from '../../../assets/story-assets/5_white.svg'
export { default as img5Disabled } from '../../../assets/story-assets/5_greyed.svg'
