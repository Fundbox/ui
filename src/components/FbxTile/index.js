import FbxTile from './FbxTile.vue'
import { registerComponents, vueUse } from '../../utils/plugins'

const components = {
  FbxTile
}

const VuePlugin = {
  install(Vue) {
    registerComponents(Vue, components)
  }
}

vueUse(VuePlugin)

export default VuePlugin
export * from './models/Tile'
