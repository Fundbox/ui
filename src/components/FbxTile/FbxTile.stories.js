import { Tile } from '../FbxTileGrid/Tile'
import { boolean as booleanKnob, object as objectKnob, text as textKnob } from '@storybook/addon-knobs'
import * as images from './story-assets'
import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'
import FbxTile from './FbxTile.vue'
import summary from './FbxTile.md'

const stories = storiesOf('Components/Tile', module)
const withSummary = withInfo({ summary })

function defaultStory() {
  // Knobs:
  const tileName = textKnob('Tile Title:', 'WellsFargo')
  const imageNormal = textKnob('Image normal:', images.img5Normal)
  const imageHover = textKnob('Image hover:', images.img5Hover)
  const imageDisabled = textKnob('Image disabled:', images.img5Disabled)
  const hoverColor = textKnob('Tile hover color:')
  const isEnabled = booleanKnob('Tile is enabled:', true)
  const isConnected = booleanKnob('Tile is connected:', false)
  const metadata = objectKnob('Metadata:', {})

  return {
    components: { FbxTile },
    data() {
      return {
        tileData: new Tile(tileName, imageNormal, imageHover, imageDisabled, hoverColor, isEnabled, isConnected, metadata),
        containerStyle: {
          height: '100px',
          width: '150px',
          background: '#eeeeee',
          overflow: 'auto',
          padding: '15px',
          resize: 'both',
        },
      }
    },
    template: `
      <div :style="containerStyle">
        <fbx-tile :tile="tileData"></fbx-tile>
      </div>
    `,
  }
}

stories.add('default', withSummary(defaultStory))
