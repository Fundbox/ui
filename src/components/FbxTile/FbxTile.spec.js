import { shallowMount } from '@vue/test-utils'
import FbxTile from './FbxTile.vue'

describe('Components/FbxTile', () => {
  describe('snapshots', () => {
    it('renders the default correctly', () => {
      const wrapper = shallowMount(FbxTile, {
        // mounting options https://vue-test-utils.vuejs.org/api/options.html
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
