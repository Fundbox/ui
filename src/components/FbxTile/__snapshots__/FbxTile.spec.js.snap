// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxTile snapshots renders connected correctly 1`] = `<div class="tile connected"><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled" style="display: none;"></div>`;

exports[`Components/FbxTile snapshots renders connected disabled correctly 1`] = `<div class="tile disabled connected"><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal" style="display: none;"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled"></div>`;

exports[`Components/FbxTile snapshots renders disabled correctly 1`] = `<div class="tile disabled"><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal" style="display: none;"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled"></div>`;

exports[`Components/FbxTile snapshots renders disabled hover correctly 1`] = `<div class="tile disabled"><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal" style="display: none;"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled"></div>`;

exports[`Components/FbxTile snapshots renders disabled hover correctly 2`] = `<div class="tile disabled"><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal" style="display: none;"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled"></div>`;

exports[`Components/FbxTile snapshots renders hover correctly 1`] = `<div class="tile" style="background: red;"><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled" style="display: none;"></div>`;

exports[`Components/FbxTile snapshots renders hover correctly 2`] = `<div class="tile" style=""><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled" style="display: none;"></div>`;

exports[`Components/FbxTile snapshots renders placeholder correctly 1`] = `
<div class="tile placeholderTile">
  <!---->
</div>
`;

exports[`Components/FbxTile snapshots renders the default correctly 1`] = `<div class="tile"><img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="normal"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="hover"> <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO 9TXL0Y4OHwAAAABJRU5ErkJggg==" alt="tileName" class="disabled" style="display: none;"></div>`;
